phlib/treenew.c

```diff
@@ -5576,10 +5576,36 @@ VOID PhTnpDrawCell(
     RECT textRect; // working rectangle, modified as needed
     ULONG textFlags; // DT_* flags
     LONG iconVerticalMargin; // top/bottom margin for icons (determined using height of small icon)
+    static HFONT fixedWidthFont = NULL;
 
     font = Node->Font;
     textFlags = Column->TextFlags;
 
+    if (Column->Alignment & PH_TREENEW_FIXED_WIDTH_FONT)
+    {
+        if (fixedWidthFont == NULL)
+        {
+            fixedWidthFont = CreateFontA(
```

Comment:
With the caveat that I am _not_ a UI guy 😄: 

Take a look at `PhDuplicateFont` and `PhDuplicateFontWithNewWeight`. It might be possible to duplicate the current font and specify `FF_MODERN`? Or maybe just use `PhCreateFont`L"Consolas", ...);` with the same weight and height as the current font?

Reply:
Maybe also a problem of double-assign of the `static` here? Or perhaps the UI thread here is guaranteed to be serialized already (my lack of UI experience is showing).

Reply:
> Maybe also a problem of double-assign of the `static` here? Or perhaps the UI thread here is guaranteed to be serialized already (my lack of UI experience is showing).

I did write it with the assumption that it'd either be single-threaded for UI sections or that a duplicate font handle would be a negligible memory leak. Of course it was even less negligible previously when it was `GetStockObject(SYSTEM_FIXED_WIDTH)` which presumably uses some sort of object interning or static handles since `GetStockObject` says `"It is not necessary (but it is not harmful) to delete stock objects by calling DeleteObject."`

Also, I'll make a commit for the review comments sometime later today/currenttime+12h or something

Reply:
Thanks for going the extra mile here. We'll chat more later.

It may also make sense to make the "monospace" font a global that gets initialized like the other fonts the application uses. Then possibly add the ability for users to choose/configure the font like the others are. This will make it easier to drop monospace into other areas too.